Public/Src/FrontEnd/Core/FrontEndHostController.cs

```diff
@@ -1154,14 +1155,15 @@ public bool TryInitializeFrontEndsAndResolvers(IConfiguration configuration, Qua
                     return false;
                 }
 
-                // TODO: Make initialization async.
-                if (!resolver.InitResolverAsync(resolverConfiguration, maybeWorkspaceResolver.Result).GetAwaiter().GetResult())
-                {
-                    // Error has been reported by the corresponding front-end.
-                    return false;
-                }
-
                 resolvers.Add(resolver);
+                initResolverTasks[i] = resolver.InitResolverAsync(resolverConfiguration, maybeWorkspaceResolver.Result);
+            }
+
+            var results = await TaskUtilities.SafeWhenAll(initResolverTasks);
+            if (results.Any(r => !r))
+            {
+                // Error should have been reported.
+                return false;
             }

```

Comment:
What motivated the change to make the resolvers async? If it was perf for allowing them to init in parallel, what are the results?

Reply:
That's what motivated it.  The results are inconclusive.  I know we sometimes see long resolver initialization times, but that can be due to NugetResolver and DownloadResolver downloading stuff from the internet.  With this change, download times are not going to improve, but in some cases more parallelism might speed things up a bit

Reply:
I concur. And in my own personal experience the download resolver is what's going on most of the time. I guess the problem is that there is no progress indication on that case, and you just see 'evaluating blah...'

---
In reply to: [345330786](https://github.com/microsoft/BuildXL/pull/1197#discussion_r345330786) [](ancestors = 345330786)